File: StemBranch.java
package stembranch;

import java.util.Scanner;

public class StemBranch {
	public static String stembranch(int year) {
		String[] stem = { "甲", "乙", "丙", "丁", "戊", "己", "庚", "辛", "壬", "癸" };
		String[] branch = { "子", "丑", "寅", "卯", "辰", "巳", "午", "未", "申", "酉", "戌", "亥" };
		return stem[(year - 4) % 10] + branch[(year - 4) % 12];
	}

	public static void main(String[] args) {
		Scanner scanner = new Scanner(System.in);
		int year = scanner.nextInt();
		scanner.close();
		System.out.println(stembranch(year));
	}
}

File: stembranch.js
let stem = "甲乙丙丁戊己庚辛壬癸";
let branch = "子丑寅卯辰巳午未申酉戌亥";


function stembranch(year) {
    return stem[(year - 4) % 10] + branch[(year - 4) % 12];
}


class StemBranch {
    stem = "甲乙丙丁戊己庚辛壬癸";
    branch = "子丑寅卯辰巳午未申酉戌亥";
    constructor() {

    }

    stembranch(year) {
        return this.stem[(year - 4) % 10] + this.branch[(year - 4) % 12];
    }
}
